src/sas/qtgui/Plotting/SlicerParameters.py

```diff
@@ -153,47 +201,122 @@ def onGeneratePlots(self, isChecked):
+        self.cbFitOptions.setEnabled(enabled)
 
     def onApply(self):
         """
         Apply current slicer to selected plots
         """
+        plots = []
         for row in range(self.lstPlots.count()):
             item = self.lstPlots.item(row)
-            isChecked = item.checkState() == QtCore.Qt.Checked
+            isChecked = item.checkState() != QtCore.Qt.Unchecked
             # Only checked items
             if not isChecked:
                 continue
             plot = item.text()
-            # don't assign to itself
-            if plot == self.parent.data[0].name:
-                continue
-            # a plot might have been deleted
-            if plot not in self.active_plots:
-                continue
-            # get the plotter2D instance
-            plotter = self.active_plots[plot]
-            # Assign model to slicer
-            index = self.cbSlicer.currentIndex()
-            slicer = self.callbacks[index]
-            plotter.setSlicer(slicer=slicer)
-            # override slicer model
-            plotter.slicer._model = self.model
-            # force conversion model->parameters in slicer
-            plotter.slicer.setParamsFromModel()
-        pass
+            # Apply plotter to a plot
+            self.applyPlotter(plot)
+            # Save 1D plots if required
+            plots.append(plot)
+        if self.isSave:
+            self.save1DPlotsForPlot(plots)
+        pass  # debug anchor
+
+    def applyPlotter(self, plot):
+        """
+        Apply the current slicer to a plot
+        """
+        # don't assign to itself
+        if plot == self.parent.data[0].name:
+            return
+        # a plot might have been deleted
+        if plot not in self.active_plots:
+            return
+        # get the plotter2D instance
+        plotter = self.active_plots[plot]
+        # Assign model to slicer
+        index = self.cbSlicer.currentIndex()
+
+        slicer = self.callbacks[index]
+        if slicer is None:
+            return
+        plotter.setSlicer(slicer=slicer, reset=False)
+        # override slicer model
+        plotter.slicer._model = self.model
+        # force conversion model->parameters in slicer
+        plotter.slicer.setParamsFromModel()
+
+    def save1DPlotsForPlot(self, plots):
+        """
+        Save currently shown 1D sliced data plots for a given 2D plot
+        """
+        items_for_fit = []
+        for plot in plots:
+            for item in self.active_plots.keys():
+                data = self.active_plots[item].data[0]
+                if not isinstance(data, Data1D):
+                    continue
+                if plot not in data.name:
+                    continue
+                filename = data.name if self.txtExtension.text() == ""\
+                    else data.name + "_" + str(self.txtExtension.text())
+                # Saved as "TXT" file, as in original impl
+                filename_ext = filename + ".txt"
```

Comment:
Potential improvement (not necessarily here): Allow data to be saved as NXcanSAS or cansas XML.

Reply:
Considering we give this option to all other 1D plots, this needs to be implemented here as well.